http-filter-example/http_filter.cc

```diff
@@ -26,9 +29,45 @@ const std::string HttpSampleDecoderFilter::headerValue() const {
   return config_->val();
 }
 
+int HttpSampleDecoderFilter::headerExtra() const {
+  return config_->extra();
+}
+
 FilterHeadersStatus HttpSampleDecoderFilter::decodeHeaders(RequestHeaderMap& headers, bool) {
-  // add a header
-  headers.addCopy(headerKey(), headerValue());
+
+  // parse config: key is header operation, value is a vector of arguments for the operation
+  std::unordered_map<std::string, std::vector<std::string>> header_ops;
+  std::string header_config = headerValue();
+
+  // Find the position of the first space character
+  size_t spacePos = header_config.find(' ');
+  std::string operation = header_config.substr(0, spacePos);
+  std::string args = header_config.substr(spacePos + 1);
+
+  // Create a string stream to iterate over the arguments
+  std::istringstream iss(args);
+  std::vector<std::string> values;
```

Comment:
Nitpick: I think this can be a `vector<string_view>` to avoid copying. But if you move the parsing into the init, you'll have to be careful the views don't outlive the original. Maybe you could pass the original string the views are into with the values vector; or maybe it's easier to just copy the substrings